dndserver/handlers/lobby.py

```diff
@@ -74,3 +80,27 @@ def open_map_select(ctx, msg):
     req.ParseFromString(msg)
     res = SS2C_OPEN_LOBBY_MAP_RES()
     return res
+
+
+def auto_match(ctx, msg):
+    """Occurs when the client attempts to find a match"""
+    req = SC2S_AUTO_MATCH_REG_REQ()
+    req.ParseFromString(msg)
+    party = get_party(account_id=sessions[ctx.transport].account.id)
+    matchteam = SS2C_AUTO_MATCH_REG_TEAM_NOT(result=pc.SUCCESS, mode=req.mode)
+    header = make_header(matchteam)
+    if req.mode == SC2S_AUTO_MATCH_REG_REQ.MODE.REGISTER:
+        matchmaking_users.append({"party": party, "difficulty": req.mode})
+    elif req.mode == SC2S_AUTO_MATCH_REG_REQ.MODE.CANCEL:
+        try:
+            matchmaking_users.remove({"party": party, "difficulty": req.mode})
+        except ValueError:
+            pass
+    if len(party.players) > 1:
+        for user in party.players:
+            transport, _ = get_user(account_id=user.account.id)
+            if req.mode == SC2S_AUTO_MATCH_REG_REQ.MODE.REGISTER:
+                transport.write(header + matchteam.SerializeToString())
+            elif req.mode == SC2S_AUTO_MATCH_REG_REQ.MODE.CANCEL:
+                transport.write(header + matchteam.SerializeToString())
```

Comment:
The if statement does not do anything here. You can replace it with:
```suggestion
            transport.write(header + matchteam.SerializeToString())
```